tests/test_date_conversions.py

```diff
@@ -55,8 +56,50 @@ def test_return_periods(requests_mock, flask_test_client):
     assert output_3["from-year"] == ["2022/2023", "2023/2024"]
 
 
-def test_quarter_to_dates():
-    assert quarter_to_date(quarter="2", year="2022/2023") == "2022-07-01T00:00:00Z"
-    assert quarter_to_date(quarter="1", year="2020/2021") == "2020-04-01T00:00:00Z"
-    assert quarter_to_date(quarter="3", year="2019/2020") == "2019-10-01T00:00:00Z"
-    assert quarter_to_date(quarter="4", year="2021/2022") == "2021-01-01T00:00:00Z"
+def test_financial_quarter_from_mapping():
+    # Q1 includes April, May, June
+    assert (
+        financial_quarter_from_mapping(quarter="1", year="2020/2021")
+        == "2020-04-01T00:00:00Z"
+    )
+
+    # Q2 includes July, August, September
+    assert (
+        financial_quarter_from_mapping(quarter="2", year="2022/2023")
+        == "2022-07-01T00:00:00Z"
+    )
+
+    # Q3 includes October, November, December
+    assert (
+        financial_quarter_from_mapping(quarter="3", year="2019/2020")
+        == "2019-10-01T00:00:00Z"
+    )
+
+    # January, February, March - selecting Q4 should return up to January of the next year
+    assert (
+        financial_quarter_from_mapping(quarter="4", year="2021/2022")
+        == "2022-01-01T00:00:00Z"
+    )
+    assert (
+        financial_quarter_from_mapping(quarter="4", year="2020/2021")
+        == "2021-01-01T00:00:00Z"
+    )
+
+
+def test_financial_quarter_to_mapping():
+    assert (
+        financial_quarter_to_mapping(quarter="1", year="2023/2024")
+        == "2023-06-30T00:00:00Z"
+    )
+    assert (
+        financial_quarter_to_mapping(quarter="2", year="2023/2024")
+        == "2023-09-30T00:00:00Z"
+    )
+    assert (
+        financial_quarter_to_mapping(quarter="3", year="2023/2024")
+        == "2023-12-31T00:00:00Z"
+    )
+    assert (
+        financial_quarter_to_mapping(quarter="4", year="2023/2024")
+        == "2024-03-31T00:00:00Z"
```

Comment:
can we add a test to check the "range" of dates is as expected? Can probs do for just one quarter, ie pass the same year/quarter to both mapping functions in one test. And assert that you have a date range of 3 months?